closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -198,6 +271,99 @@ public static boolean isProviderAllowed(Class<?> providerClazz) {
         return true;
     }
 
+    /**
+     * Figure out the full profile ID.
+     *
+     * Use the default or user selected profile and attempt to find
+     * an appropriate entry in the java.security properties.
+     *
+     * If a profile cannot be found, or multiple defaults are discovered
+     * for a single profile,an appropriate message is printed and the
+     * system exits.
+     *
+     * @param props the java.security properties
+     */
+    private static void getProfileID(Properties props) {
+        String potentialProfileID = "RestrictedSecurity." + selectedProfile;
+
+        if (selectedProfile.indexOf(".") != -1) {
+            /* The default profile is used, or the user specified the
+             * full <profile.version>.
+             */
+            if (debug != null) {
+                debug.println("Profile specified using full name (i.e., <profile.version>).");
+            }
+            for (Object keyObject : props.keySet()) {
+                // A matching property is found.
+                if (keyObject instanceof String key) {
+                    if (key.startsWith(potentialProfileID)) {
+                        profileID = potentialProfileID;
+                        return;
+                    }
+                }
+            }
+            printStackTraceAndExit(selectedProfile + " is not present in the java.security file.");
+        } else {
+            /* The user specified the only the <profile> without
+             * indicating the <version> part.
+             */
+            if (debug != null) {
+                debug.println("Profile specified without version (i.e., <profile>).");
+            }
+            String defaultMatch = null;
+            for (Object keyObject : props.keySet()) {
+                // A matching property is found.
+                if (keyObject instanceof String key) {
+                    if (key.startsWith(potentialProfileID)) {
+                        // Find the property for specifying if this is the default profile.
+                        if (key.endsWith(".desc.default")) {
+                            // Check if property is set to true.
+                            if (Boolean.parseBoolean(props.getProperty(key))) {
+                                // Check if multiple defaults exist and act accordingly.
+                                if (defaultMatch == null) {
+                                    defaultMatch = key.split("\\.desc")[0];
+                                } else {
+                                    printStackTraceAndExit("Multiple default profiles for " + selectedProfile);
```

Comment:
All these messages can be more explicit and contain "Restricted Security" or similar, some of them already do.
i.e. "Multiple Restricted Security default profiles for ".
We know the context while implementing and testing, but it may not be clear to a user, in particular if the command line options are not explicitly set by the user but inherited from somewhere.

Reply:
Added references to `RestrictedSecurity` for clarity.